aeron-driver/src/main/java/io/aeron/driver/MediaDriver.java

```diff
@@ -522,23 +525,23 @@ public void close()
             {
                 isClosed = true;
 
-                CloseHelper.close(logFactory);
+                AeronCloseHelper.close(errorHandler, logFactory);
 
-                CloseHelper.close(lossReport);
+                AeronCloseHelper.close(errorHandler, lossReport);
                 IoUtil.unmap(lossReportBuffer);
                 lossReportBuffer = null;
 
                 if (errorHandler instanceof AutoCloseable)
                 {
-                    CloseHelper.close((AutoCloseable)errorHandler);
+                    CloseHelper.quietClose((AutoCloseable)errorHandler); // Ignore error to ensure the rest is closed
                 }
 
                 IoUtil.unmap(cncByteBuffer);
                 cncByteBuffer = null;
 
                 if (dirDeleteOnShutdown && null != aeronDirectory())
                 {
-                    deleteAeronDirectory();
+                    CloseHelper.quietClose(this::deleteAeronDirectory);
```

Comment:
Delete can be better handled with this. https://github.com/real-logic/agrona/commit/43cdebd5dfc0d1daa2684da632de951f4fecae55

Reply:
@mjpt777 Have you seen my [previous comment](https://github.com/real-logic/aeron/pull/840#discussion_r376727260)? We won't be able to use new API here, because we are deleting Aeron directory which contains the error log thus we cannot use `ErrorHandler` here.
Plus it would require bypassing `deleteAeronDirectory` and calling the `IoUtils#delete` method directly.

In this particular case we can probably just call `deleteAeronDirectory` without `quietClose` since the only other action in the close sequence is to call `super.close` which is an empty method as of now.

Reply:
You are right that we have an ordering issue here. We cannot `quietClose` this otherwise we miss cases on Windows that should be notified.

Reply:
@mjpt777 I changed code to call `deleteAeronDirectory()` directly.